File: Java/22_BitManipulation/LC_260_SingleNumberIII.java
class Solution {
    public int[] singleNumber(int[] nums) {
        int sum = 0;
        for (int num : nums) {
            sum ^= num;
        }

        int lbs = sum & (-sum);

        int num1 = 0;
        int num2 = 0;
        for (int num : nums) {
            if ((num & lbs) != 0) {
                num1 ^= num;
            } else {
                num2 ^= num;
            }
        }

        return new int[] { num1, num2 };
    }
}

public class LC_260_SingleNumberIII {

}

File: Swift/Leetcode.playground/Sources/22_BitManipulation/LC_260_SingleNumberIII.swift
class Solution {
    func singleNumber(_ nums: [Int]) -> [Int] {
        var sum = 0
        for num in nums {
            sum ^= num
        }
        let lbs = sum & (-sum)
        var num1 = 0
        var num2 = 0
        for num in nums {
            if (num & lbs) != 0 {
                num1 ^= num
            } else {
                num2 ^= num
            }
        }
        return [num1, num2]
    }
}